### PR TITLE
statistics: prevent columns not present in the information schema from affecting the final result

### LINKED ISSUE #55713: master branch pull-lightning-integration-test failed
https://do.pingcap.net/jenkins/blue/organizations/jenkins/pingcap%2Ftidb%2Fpull_lightning_integration_test/detail/pull_lightning_integration_test/1193/pipeline/306

```

*************************** 1. row ***************************

count(i): 5000

  sum(i): 17502500

TEST FAILED: OUTPUT DOES NOT CONTAIN 'Row_count: 5000'

____________________________________

[Wed Aug 28 11:54:09 CST 2024] Executing SQL: SHOW STATS_META WHERE Table_name = 'tbl';

^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

script returned exit code 1
```

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close #55713

Problem Summary:

### What changed and how does it work?

Sometimes, during async loading, the information for this column may not be retrieved. In such cases, it is crucial to return promptly; otherwise, it could result in corrupted data.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)

tested by internal E2E test, it is succeed.

- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by lance6716 — APPROVED** [review]

File: pkg/statistics/handle/storage/read.go [hunk 1]
```diff
@@ -625,6 +625,10 @@ func loadNeededColumnHistograms(sctx sessionctx.Context, statsHandle statstypes.
 			return nil
 		}
 		colInfo = tblInfo.Meta().GetColumnByID(col.ID)
+		if colInfo == nil {
```
Comment (D3Hunter) [inline pkg/statistics/handle/storage/read.go:628]: can you change the PR title, seems we are fixing some corner case of stats gathering

Comment (hawkingrei) [reply pkg/statistics/handle/storage/read.go:628]: OK，Updated.

**Review by D3Hunter — APPROVED** [review]

**Comment by hawkingrei** [comment]
/hold

Waiting for the E2E test‘s result.

**Review by lance6716 — COMMENTED** [review]
image

/unhold

**Comment by hawkingrei** [comment]
/unhold

tested by internal E2E test, it is succeed.

**Review by AilinKid — APPROVED** [review]

**Pull request merged**